pkg/handler/iam_handler.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// handler package that handles IAM request and IAM policy updates.
+package handler
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/iam/apiv1/iampb"
+	"github.com/abcxyz/access-on-demand/apis/v1alpha1"
+	"github.com/googleapis/gax-go/v2"
+	"github.com/sethvargo/go-retry"
+	"google.golang.org/genproto/googleapis/type/expr"
+
+	resourcemanager "cloud.google.com/go/resourcemanager/apiv3"
+)
+
+// ConditionTitle of IAM bindings added by AOD.
+var ConditionTitle = "abcxyz-aod-expiry"
+
+// IAMHandler updates IAM policies of GCP organizations, folders, and projects
+// based on the IAM request received.
+type IAMHandler struct {
+	organizationsClient *resourcemanager.OrganizationsClient
+	foldersClient       *resourcemanager.FoldersClient
+	projectsClient      *resourcemanager.ProjectsClient
```

Comment:
Here you are still depending on the implementation. Instead, I think they should be of type `IAMClient` (your interface)